cmd/fleet/calendar_cron.go

```diff
@@ -195,10 +195,13 @@ func processCalendarFailingHosts(
 
 	const consumers = 20
 	hostsCh := make(chan fleet.HostPolicyMembershipData)
-	g, ctx := errgroup.WithContext(ctx)
+	var wg sync.WaitGroup
 
 	for i := 0; i < consumers; i++ {
-		g.Go(func() error {
+		wg.Add(+1)
+		go func() {
+			defer wg.Done()
+
 			for host := range hostsCh {
```

Comment:
Is it guaranteed that only 1 goroutine receives the host, and is it going to be spread out across the goroutines?

Reply:
Yes, due to the previous call to `filterHostsWithSameEmail`. Same thing on the `removeCalendarEventsFromPassingHosts` (there's filtering that makes it so that such slices have one item per email).